### PR TITLE
Travis config for creating and publishing multi arch images

### PR DESCRIPTION
**Pull request by aishwaryabk**
Signed-off-by: Aishwarya Kamat <aishwarya.kamat@ibm.com>

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
+os: linux
+dist: bionic
+language: bash
+services: docker
+
+branches:
+  only:
+  - devel
```
Comment (yussufsh) [inline .travis.yml:8]: We can have this only on `master` and run only on pushing tags.

Comment (aishwaryabk) [reply .travis.yml]: Updated the change.

Comment (yussufsh) [reply .travis.yml]: I do not see on event tag. The images should be built and pushed only when a tag is pushed.

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+env:
+  - VERSION=4.7
+
+jobs:
+  include:
+    - stage: build x86_64 container image
+      arch: amd64
+      script:
+        - docker build --build-arg RELEASE_VER=$VERSION -t openshift-install-powervs:v1.1.2-ocp$VERSION-$(uname -m) -f images/Dockerfile .
```
Comment (yussufsh) [inline .travis.yml:18]: The version should not be hardcoded.... v1.1.2 should be the tag version which we are going to push.

**Comment by yussufsh** [comment]
@aishwaryabk We can also push `openshift-install-powervs:ocp$VERSION` (without the tag version) which will be always the `latest` image version.

Courtesy @bpradipt

**Comment by aishwaryabk** [comment]
> @aishwaryabk We can also push `openshift-install-powervs:ocp$VERSION` (without the tag version) which will be always the `latest` image version.
> 
> Courtesy @bpradipt

Thanks! Updated the changes.

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+      arch: ppc64le
+      script:
+        - docker build --build-arg RELEASE_VER=$VERSION -t openshift-install-powervs:$TRAVIS_TAG-ocp$VERSION-$(uname -m) -f images/Dockerfile .
+        - docker login -u $DOCKER_USERNAME -p $DOCKER_PASSWORD
+        - docker push openshift-install-powervs:$TRAVIS_TAG-ocp$VERSION-$(uname -m) 
+    - stage: Build multi-arch image
+      script: 
+        - export DOCKER_CLI_EXPERIMENTAL=enabled
+        - docker manifest create openshift-install-powervs:$TRAVIS_TAG-ocp$VERSION --amend openshift-install-powervs:$TRAVIS_TAG-ocp$VERSION-x86_64 --amend openshift-install-powervs:$TRAVIS_TAG-ocp$VERSION-ppc64le
+        - docker manifest create openshift-install-powervs:ocp$VERSION --amend openshift-install-powervs:$TRAVIS_TAG-ocp$VERSION-x86_64 --amend openshift-install-powervs:$TRAVIS_TAG-ocp$VERSION-ppc64le
```
Comment (yussufsh) [inline .travis.yml:33]: Can we use docker tag since it is the same image, instead of creating manifest again?

File: .travis.yml [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+  - VERSION=4.5
+  - VERSION=4.6
+  - VERSION=4.7
+
+jobs:
+  include:
+    - stage: Build x86_64 image
+      arch: amd64
+      script:
+        - docker build --build-arg RELEASE_VER=$VERSION -t openshift-install-powervs:$TRAVIS_TAG-ocp$VERSION-$(uname -m) -f images/Dockerfile .
```
Comment (yussufsh) [inline .travis.yml:20]: Please use the `https://quay.io/repository/powercloud/` org for the image and login to quay.io.

**Review by yussufsh — CHANGES_REQUESTED** [review]

**Review by yussufsh — COMMENTED** [review]
@aishwaryabk we need this job to run only when a tag is pushed.. not for each PR/commit.

File: .travis.yml (hunk 3 above)
Comment (aishwaryabk) [reply .travis.yml]: docker tag cannot be used on the manifest list since 'docker manifest create' creates a list and not a docker image.

**Review by Prajyot-Parab — APPROVED** [review]
/lgtm

**Review by bpradipt — APPROVED** [review]
/lgtm

**Pull request merged**